tests/validation/tests/v3_api/test_node_label.py

```diff
@@ -0,0 +1,1178 @@
+    cluster_node_template["test_label"] = test_label
+    cluster_node_template["label_value"] = label_value
+
+    def fin():
+        client = get_user_client()
+        cluster = cluster_node_template["cluster"]
+        if cluster is not None:
+            node_pools_list = client.list_nodePool(clusterId=cluster.id).data
+            # get unique node template ids
+            unique_node_pool = {}
+            for node_pool in node_pools_list:
+                if node_pool.nodeTemplateId not in unique_node_pool.keys():
+                    unique_node_pool[node_pool.nodeTemplateId] = \
+                        client.list_node_template(
+                            id=node_pool.nodeTemplateId).data[0]
+            print("unique_node_pool: ", unique_node_pool)
+            client.delete(cluster_node_template["cluster"])
+            wait_for_cluster_delete(client, cluster["name"])
+            time.sleep(10)
+            for key, value in unique_node_pool.items():
+                client.delete(value)
+        if cluster_custom["cluster"] is not None:
+            client.delete(cluster_custom["cluster"])
+        if cluster_custom["aws_node"] is not None:
+            delete_node(cluster_custom["aws_node"])
+        if custom_cluster_add_edit["cluster"] is not None:
+            client.delete(custom_cluster_add_edit["cluster"])
+        if custom_cluster_add_edit["aws_node"] is not None:
+            print("custom_cluster_add_edit_nodes: ",
+                  custom_cluster_add_edit["aws_node"])
+            delete_node(custom_cluster_add_edit["aws_node"])
+
+    request.addfinalizer(fin)
+
+
+def check_cluster_deleted(client):
+    def _find_condition(resource):
+        cluster = client.reload(resource)
+        if len(cluster["data"]) == 0:
+            return True
+        else:
+            return False
+
+    return _find_condition
+
+
+def check_label_added(test_label):
+    def _find_condition(resource):
+        node_labels = resource.labels.data_dict()
+
+        if test_label in node_labels:
+            return True
+        else:
+            return False
+
+    return _find_condition
+
+
+def check_label_removed(test_label):
+    def _find_condition(resource):
+        node_labels = resource.labels.data_dict()
+
+        if test_label not in node_labels:
+            return True
+        else:
+            return False
+
+    return _find_condition
+
+
+def validate_label_set_on_node(client, node, test_label, label_value):
+    """
+    This method checks if the label is added on the node via API and kubectl
+    :param client: user client
+    :param node: node on which user has to validate if the label is added
+    :param test_label: Label to be validated on the node
+    :param label_value: label value to be checked
+    :return: None
+    """
+    print("label_value: ", label_value)
+    # check via API
+    node = client.reload(node)
+    node_labels = node.labels.data_dict()
+    print("three: ", node_labels)
```

Comment:
three ?

Reply:
This was for my testing purpose :). will delete this